idp_task1.cpp

```diff
@@ -4,52 +4,59 @@
 static const int LISTENQ = 16;
 static const int DEFAULT_ERROR = -1;
 static const int SERVER_PORT = 8877;
-static const int ECHO_STR_LEN = 6;                                            
+static const char* echo_str_prefix = "echo: ";
+static const int ECHO_STR_LEN = strlen(echo_str_prefix);
 
 void createTestTCPServer()
 {
     try 
     {
-    std::unique_ptr<SocketWrapper> server_socket = std::make_unique<SocketWrapper>(AF_INET, SOCK_STREAM, 0, SERVER_PORT);
+        SocketWrapper_uptr server_socket = std::make_unique<SocketWrapper>(AF_INET, SOCK_STREAM, 0, SERVER_PORT);
 
-    server_socket->Bind();
+        sockaddr_in server_socket_address;
+        bzero(&server_socket_address, sizeof(sockaddr_in));
+        server_socket_address.sin_family = AF_INET;
+        server_socket_address.sin_port = htons (SERVER_PORT);
+        server_socket_address.sin_addr.s_addr = INADDR_ANY ; 
 
-    server_socket->Listen(LISTENQ);
+        server_socket->Bind(&server_socket_address);
```

Comment:
Not exactly what I've suggested.
All we need to bind the socket to specific address is the pair of address and port. From the UX point of view it is much better to pass only these two params directly to `Bind` function, and `Bind` will know what to do with them.
We also need an address family here, but in fact no, we don't need it, 'cause we've already created socket of specific type (family). So to optimize class interface here (and so UX too), we have two ways how to not pass AF parameter to `bind`:
1. store AF to variable when we're creating socket (we can use `sockaddr_in` structure already existing in our class)
2. get created socket AF using `getsockopt(...)` function
Note that second way might be non-portable (Unix-specific).